server/gameconnection.py

```diff
@@ -311,19 +317,29 @@ async def handle_operation_complete(self, army, secondary, delta):
             )
             row = await result.fetchone()
             if not row:
-                self._logger.debug("can't find coop map: %s", self.game.map_file_path)
+                self._logger.debug(
+                    "can't find coop map: %s", self.game.map_file_path
+                )
                 return
             mission = row["id"]
 
-            await conn.execute(
-                coop_leaderboard.insert().values(
-                    mission=mission,
-                    gameuid=self.game.id,
-                    secondary=secondary,
-                    time=delta,
-                    player_count=len(self.game.players),
+            previously_completed_operations = await conn.execute(
```

Comment:
No reason to do a database query here, just use a boolean value on the `self.game` to keep track of whether or not the entry has been written yet.

Reply:
Oh nice idea, thank you!